lncfg/db.go

```diff
@@ -391,6 +412,89 @@ func (db *DB) GetBackends(ctx context.Context, chanDBPath,
 			Remote:     true,
 			CloseFuncs: closeFuncs,
 		}, nil
+
+	case SqliteBackend:
+		sqliteBackend, err := kvdb.Open(
+			kvdb.SqliteBackendName, ctx, db.Sqlite, LndDBName,
+			NSChannelDB,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("error opening sqlite graph "+
+				"DB: %v", err)
+		}
+		closeFuncs[NSChannelDB] = sqliteBackend.Close
+
+		sqliteMacaroonBackend, err := kvdb.Open(
+			kvdb.SqliteBackendName, ctx, db.Sqlite, LndDBName,
+			NSMacaroonDB,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("error opening sqlite "+
+				"macaroon DB: %v", err)
+		}
+		closeFuncs[NSMacaroonDB] = sqliteMacaroonBackend.Close
+
+		sqliteDecayedLogBackend, err := kvdb.Open(
```

Comment:
Realized that we aren't storing the neutrino data in here. Current entrypoint is in `initNeutrinoBackend`. It actually still uses the `walletdb` directly also vs `kvdb`.

Reply:
I used this diff to have the neutrino data be stored in sqlite as well:
```
diff --git a/config_builder.go b/config_builder.go
index 6f427ffd2..f0d425f1a 100644
--- a/config_builder.go
+++ b/config_builder.go
@@ -1190,13 +1190,28 @@ func initNeutrinoBackend(cfg *Config, chainDir string,
 		return nil, nil, err
 	}
 
-	dbName := filepath.Join(dbPath, "neutrino.db")
-	db, err := walletdb.Create(
-		"bdb", dbName, !cfg.SyncFreelist, cfg.DB.Bolt.DBTimeout,
+	var (
+		db  walletdb.DB
+		err error
 	)
-	if err != nil {
-		return nil, nil, fmt.Errorf("unable to create neutrino "+
-			"database: %v", err)
+	switch {
+	case cfg.DB.Backend == kvdb.SqliteBackendName:
+		db, err = kvdb.Open(
+			kvdb.SqliteBackendName, context.Background(), cfg.DB.Sqlite,
+			"neutrino.db", "neutrino",
+		)
+		if err != nil {
+			return nil, nil, err
+		}
+	default:
+		dbName := filepath.Join(dbPath, "neutrino.db")
+		db, err = walletdb.Create(
+			"bdb", dbName, !cfg.SyncFreelist, cfg.DB.Bolt.DBTimeout,
+		)
+		if err != nil {
+			return nil, nil, fmt.Errorf("unable to create neutrino "+
+				"database: %v", err)
+		}
 	}
 
 	headerStateAssertion, err := parseHeaderStateAssertion(
```

That places it into its own database (which seems like what we want to do here?).

Reply:
ok with neutrino in same file as the other DBs and running `make itest dbbackend=sqlite backend=neutrino` seems to behave ok. Just ran some of the usual flakes but looks like majority of tests pass. So perhaps fine to keep it in same file?

Reply:
To same file, or not to same file....🤔

Only arg I can think of _not_ putting it in the same file is if ppl frequently just delete that db if they want to force a resync (tho they can do so already by deleting the on disk headers). 

Looking at the way the files are set up, if we put it all in the same file (or diff file) _with_ the distinct directory, then the flat header files end up in diff a diff directory, so things end up being more scattered vs retaining our existing pretty well established file format.

Reply:
ok so even with the latest update where we _dont_ have the distinct directory?

In any case, latest update keeps the neutrino db in separate file and puts all sqlite db files in the main `data/chain/bitcoin/<network>/` dir.

Lemme know what you think